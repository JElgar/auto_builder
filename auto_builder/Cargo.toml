[package]
name = "auto_builder"
version = "0.1.0"
edition = "2021"
license = "MIT"
description = "Package for generating a 'safe' builder"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
auto_builder_derive = { path = "../auto_builder_derive", version = "0.1.0" }
darling = "0.20.5"
