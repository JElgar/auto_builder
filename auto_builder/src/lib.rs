pub use auto_builder_derive::Builder;

pub struct NoValue;
