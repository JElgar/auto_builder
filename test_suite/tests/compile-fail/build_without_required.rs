use auto_builder::Builder;

fn main() {
    #[derive(Builder)]
    struct A {
        x: i32,
        y: i32,
        z: String,
    }

    let a = A::builder().set_x(1).set_y(2).build();
}
