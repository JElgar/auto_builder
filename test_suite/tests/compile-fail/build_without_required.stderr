error[E0599]: no method named `build` found for struct `ABuilder<i32, i32, NoValue>` in the current scope
  --> tests/compile-fail/build_without_required.rs:11:44
   |
4  |     #[derive(Builder)]
   |              ------- method `build` not found for this struct
...
11 |     let a = A::builder().set_x(1).set_y(2).build();
   |             ------------ --------          ^^^^^ method not found in `ABuilder<i32, i32, NoValue>`
   |             |            |
   |             |            method `build` is available on `ABuilder<i32, NoValue, NoValue>`
   |             method `build` is available on `ABuilder<NoValue, NoValue, NoValue>`
   |
   = note: the method was found for
           - `ABuilder<i32, i32, String>`
